packages/block-editor/src/components/preview-options/index.js

```diff
@@ -56,28 +63,24 @@ export default function PreviewOptions( {
 			{ () => (
 				<>
 					<MenuGroup>
-						<MenuItem
-							className="block-editor-post-preview__button-resize"
-							onClick={ () => setDeviceType( 'Desktop' ) }
-							icon={ deviceType === 'Desktop' && check }
-						>
-							{ __( 'Desktop' ) }
-						</MenuItem>
-						<MenuItem
-							className="block-editor-post-preview__button-resize"
-							onClick={ () => setDeviceType( 'Tablet' ) }
-							icon={ deviceType === 'Tablet' && check }
-						>
-							{ __( 'Tablet' ) }
-						</MenuItem>
-						<MenuItem
-							className="block-editor-post-preview__button-resize"
-							onClick={ () => setDeviceType( 'Mobile' ) }
-							icon={ deviceType === 'Mobile' && check }
-						>
-							{ __( 'Mobile' ) }
-						</MenuItem>
+						{ devices.map( ( { type, label } ) => (
+							<MenuItem
+								key={ type }
+								className="block-editor-post-preview__button-resize"
+								onClick={ () => setDeviceType( type ) }
+								icon={ deviceType === type && check }
+							>
+								{ label }
+							</MenuItem>
+						) ) }
 					</MenuGroup>
+					<PluginPreview.Slot
```

Comment:
We should probably call this `PluginPreviewMenu` as the canvas also needs extensibility points and it'd be confusing.

Reply:
Agreed! The older PR also [had feedback](https://github.com/WordPress/gutenberg/pull/31984#issuecomment-845721683) that this name was too generic.